### PR TITLE
quadlet: fix runtime error for invalid Mount value

### PR DESCRIPTION
**Pull request by jakmeier**
If the `Mount` option inside a quadlet is missing the source=... part, the code today panics with the following message.

```
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/containers/podman/v5/pkg/systemd/quadlet.handleStorageSource(0xc000140de0?, 0x1d?, {0x0?, 0x1?}, 0x5634e39e233e?, 0x10?)
...
```

This commit checks for the missing source and returns an error to avoid the panic.

I added the check in two places, in handleStorageSource and in resolveContainerMountParams. The actual panic happens due to a unchecked index access in the former function while the porblematic configuration should already be detected before calling that function.

#### Does this PR introduce a user-facing change?

```release-note
Fix panic when a quadlet `Mount` option is missing a `source=...` value. Podman now reports a clear error instead of crashing.
```

### CODE REVIEW

**Comment by jakmeier** [comment]
Hey, first time contributor here, I just had to look in the code after hitting this bug to find out what was wrong with my quadlet. As I was already there in the code, I though I might as well submit a fix.

Let me know if I missed anything in the contributor guide or other admin stuff.

**Comment by baude** [comment]
thanks for your submission and congrats.  we appreciate any help.

the ci failed because we generally require a test for any PR submitted (or a modification of a test).  Is this something you think we can get a test for to protect against future regressions at the least?

File: pkg/systemd/quadlet/quadlet.go [hunk 1]
```diff
@@ -2058,6 +2061,9 @@ func resolveContainerMountParams(containerUnitFile, serviceUnitFile *parser.Unit
 			originalSource = val
 		}
 	}
+	if originalSource == "" {
+		return "", fmt.Errorf("missing source parameter in Mount")
+	}
```
Comment (Luap99) [inline pkg/systemd/quadlet/quadlet.go]: you handle it in handleStorageSource() so this seems redundant and is not needed
I prefer if things are validated once.

**Comment by Luap99** [comment]
Note DCO check failed here because you use a different email for the sign off compared the the git author field

> Author: Jakob Meier, Committer: Jakob Meier; Expected "Jakob Meier [mail@jakobmeier.ch](mailto:mail@jakobmeier.ch)", but got "Jakob Meier [inbox@jakobmeier.ch](mailto:inbox@jakobmeier.ch)".

We can set it manually to pass but in general these values should be the same.

**Comment by ygalblum** [comment]
> Is this something you think we can get a test for to protect against future regressions at the least?

You should add a test case where the source is empty and validate that Quadlet failed and returned the expected message. 

See other cases for example: https://github.com/containers/podman/blob/2499de2f277209e9cd4f3ce13bfc1f48472b137c/test/e2e/quadlet_test.go#L1117

**Comment by jakmeier** [comment]
Thanks everyone for the quick and helpful responses!

I've addresses all comments

- added a test case
- removed redundant check
- fixed sign off email

My new test case runs fine locally and CI seems to pass on that test, too.

I believe the remaining failing tests have nothing to do with my changes. Other PRs have the same failures.

Please take another look.

**Review by ygalblum — APPROVED** [review]
LGTM

**Review by Luap99 — APPROVED** [review]
/lgtm
